packages/esm-service-queues-app/src/queue-screen/queue-screen.component.tsx

```diff
@@ -1,15 +1,69 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { DataTableSkeleton } from '@carbon/react';
 import { useTranslation } from 'react-i18next';
 import { useActiveTickets } from './useActiveTickets';
 import PatientQueueHeader from '../patient-queue-header/patient-queue-header.component';
 import styles from './queue-screen.scss';
+import { useSelectedQueueLocationUuid } from '../helpers/helpers';
 
 interface QueueScreenProps {}
 
 const QueueScreen: React.FC<QueueScreenProps> = () => {
   const { t } = useTranslation();
-  const { activeTickets, isLoading, error } = useActiveTickets();
+  const { activeTickets, isLoading, error, mutate } = useActiveTickets();
+  const speaker = window.speechSynthesis;
+  const [isSpeaking, setIsSpeaking] = useState(false);
+  const selectedLocation = useSelectedQueueLocationUuid();
+  const locationFilteredTickets = activeTickets;
+
+  const rowData = locationFilteredTickets.map((ticket, index) => ({
+    id: `${index}`,
+    room: ticket.room,
+    ticketNumber: ticket.ticketNumber,
+    status: ticket.status,
+  }));
```

Comment:
Consider memoizing `rowData`:

```tsx
  const rowData = useMemo(() => 
    activeTickets.map((ticket, index) => ({
      id: `${index}`,
      room: ticket.room,
      ticketNumber: ticket.ticketNumber,
      status: ticket.status,
    })),
  [activeTickets]);
```

Also, is there something unique in the response that we could use as the `id` instead of `index`? Maybe a UUID?

Reply:
@denniskigen currently the response does not have uuid thats why i opted to use index